app/ui-react/syndesis/src/shared/EntityIcon.tsx

```diff
@@ -39,9 +39,11 @@ function getConnectionIcon(
     connection.icon.toLowerCase().startsWith('db:') ||
     connection.icon.toLowerCase().startsWith('extension:')
   ) {
-    return `${apiUri}/connectors/${connection.connectorId}/icon?${
-      connection.icon
-    }`;
+    if (connection.connector) {
```

Comment:
Could you maybe throw a comment in here what edge condition this `if`.`else` statement covers?  This function for sure needs plenty of supporting info IMO, each time I come in here (and probably most other folks) I've gotta remember what's going in here :-)

Reply:
The previous `connection.connectorId` property is `undefined` when connection is an extension of type `Connection`, this happens on *Create Connection* and *Connections* screen. When the `Connection` is from an `Extension` use the `connection.connector.id` to retrieve it from the `ConnectorHandler` endpoint.

Reply:
I commented on the code, see if that helps. Thanks for the review.